### PR TITLE
DMP-1450 - 2 - Retention errors

### PR DESCRIPTION
**Pull request by joshw-hmcts**
### Jira link (if applicable)

[DMP-1450](https://tools.hmcts.net/jira/browse/DMP-1450)

### Change description ###

A follow up fix to address some issues occurring in Staging:
- Mapper appended 'T00:00:00Z' to retention_date removed
- Date format changed to 'yyyy-MM-dd'
- Date object set to 12:00 to avoid issues of DST causing date to shift back to previous day
- Additional calls made after submission to refresh retention data

### Checklist

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/app/components/case/case-retention-date/case-retention-date.component.ts [hunk 1]
```diff
@@ -129,8 +129,8 @@ export class CaseRetentionDateComponent implements OnInit {
   }
 
   getOriginalRetentionDateString(rows: CaseRetentionHistory[]) {
-    const date = this.getEarliestDate(rows).retention_date;
-    return this.datePipe.transform(date, 'dd/MM/yyyy');
+    const earliestDate = this.getEarliestDate(rows).retention_date;
+    return this.datePipe.transform(earliestDate, 'dd/MM/yyyy', 'GMT+1');
```
Comment (davet1985) [inline src/app/components/case/case-retention-date/case-retention-date.component.ts:133]: Not sure about this, we shouldn't be hard-coding timezones

File: src/app/components/case/case-retention-date/case-retention-change/case-retention-change.component.ts [hunk 2]
```diff
@@ -125,7 +125,12 @@ export class CaseRententionChangeComponent implements AfterViewChecked {
 
   dateFromString(value: string) {
     // Convert UK format date string to Date object
-    return new Date(parseInt(value.split('/')[2]), parseInt(value.split('/')[1]) - 1, parseInt(value.split('/')[0]));
+    return new Date(
+      parseInt(value.split('/')[2]),
+      parseInt(value.split('/')[1]) - 1,
+      parseInt(value.split('/')[0]),
+      12
+    );
```
Comment (davet1985) [inline src/app/components/case/case-retention-date/case-retention-change/case-retention-change.component.ts:133]: Can this be done using `luxon`? Seems a little strange to need `12` at the end?

Comment (joshw-hmcts) [reply src/app/components/case/case-retention-date/case-retention-change/case-retention-change.component.ts]: Fixed

File: src/app/components/case/case-retention-date/case-retention-date.component.ts (hunk 1 above)
Comment (joshw-hmcts) [reply src/app/components/case/case-retention-date/case-retention-date.component.ts]: Removed

File: src/app/components/case/case-retention-date/case-retention-change/case-retention-change.component.ts [hunk 3]
```diff
@@ -105,8 +105,10 @@ export class CaseRententionChangeComponent implements AfterViewChecked {
         });
       }
     } else {
+      // Permanent - Add 99 years to today's date
       const date = new Date();
-      date.setDate(date.getDate() + 36159.75);
+      const ninetyNineYearsInDays = 36159.75;
+      date.setDate(date.getDate() + ninetyNineYearsInDays);
```
Comment (davet1985) [inline src/app/components/case/case-retention-date/case-retention-change/case-retention-change.component.ts:111]: Can we do this with Luxon?

```
DateTime.now().plus({years: 99})
```

Comment (joshw-hmcts) [reply src/app/components/case/case-retention-date/case-retention-change/case-retention-change.component.ts]: Changed to the above code

**Review by davet1985 — APPROVED** [review]

**Pull request merged**